shopinvader_sale_coupon/services/customer.py

```diff
@@ -12,35 +12,52 @@ def _parser_coupon(self):
         return [
             "id",
             "code",
-            "state",
             "expiration_date",
             (
                 "program_id:program",
                 [
                     "id",
                     "name",
-                    "reward_type",
-                    "reward_product_id",
-                    "reward_product_quantity",
-                    "discount_type",
-                    "discount_percentage",
-                    "discount_fixed_amount",
+                    "program_type",
+                    ("reward_ids:rewards", self._parser_coupon_reward()),
                 ],
             ),
         ]
 
+    def _parser_coupon_reward(self):
+        return [
+            "id",
+            "reward_type",
+            "discount",
+            "discount_mode",
+            "discount_applicability",
+            ("all_discount_product_ids:discount_products", self._parser_product()),
+            "reward_product_id",
+        ]
+
+    def _parser_product(self):
+        return [
+            "id",
+            "name",
+        ]
+
     def _add_coupon_info(self, partner, info):
         coupons = (
-            self.env["coupon.coupon"]
+            self.env["loyalty.card"]
             .search(
                 [
                     ("partner_id", "=", partner.id),
```

Comment:
In v16 coupons are not assigned and restricted to partners anymore. 
The problem is that coupons generated from SO's linked to the partner are not assigned to him anymore hence cannot be retrieved with this service.